test/cloudtest/test_storaged.py

```diff
@@ -49,6 +49,55 @@ def test_storaged_creation(mz: MaterializeApplication) -> None:
         storaged = f"pod/storage-{id}-0"
         wait(condition="condition=Ready", resource=storaged)
 
+        mz.environmentd.sql(f"DROP SOURCE {source}")
+        wait(condition="delete", resource=storaged)
+
+
+def test_storaged_resizing(mz: MaterializeApplication) -> None:
+    """Test that resizing a given source causes the storaged to be replaced."""
+    mz.testdrive.run_string(
+        dedent(
+            """
+            $ kafka-create-topic topic=test
+
+            $ kafka-ingest format=bytes topic=test
+            ABC
+
+            > CREATE SOURCE resize_storaged
+              FROM KAFKA BROKER '${testdrive.kafka-addr}'
+              TOPIC 'testdrive-test-${testdrive.seed}'
+              FORMAT BYTES
+              ENVELOPE NONE;
+            """
+        )
+    )
+    id = mz.environmentd.sql_query(
+        f"SELECT id FROM mz_sources WHERE name = 'resize_storaged'"
+    )[0][0]
+    assert id is not None
+    storaged = f"pod/storage-{id}-0"
+
+    wait(condition="condition=Ready", resource=storaged)
+
+    mz.testdrive.run_string(
+        dedent(
+            """
+            > ALTER SOURCE resize_storaged
+              SET (SIZE '16');
```

Comment:
This used to be a no-op with the standard storage host settings (as nothing would have changed in k8s), although I guess now with the labels something will change, right? Otherwise you might want to add some storage host sizes (as an envvar) to ensure k8s is changing something meaningful in this test.

(we avoid setting default limits for CPU and memory because we don't want to enforce limits if you are running k8s locally, as you might not have enough memory/etc)

Reply:
Yeah... even before the labels I think it would restart the pod to pick up the new args, but I'm not sure of that.

I spent a while wrestling with `kubectl wait` to watch for stuff like this and failed to get anything working -- it's pretty limited -- but handrolling a polling loop would work. If you think that's important to have, happy to include!